src/Pim/Bundle/ApiBundle/Controller/ProductModelController.php

```
@@ -187,6 +197,41 @@ public function createAction(Request $request): Response
 
     /**
      * @param Request $request
+     * @param string  $code
+     *
+     * @throws HttpException
+     *
+     * @return Response
+     */
+    public function partialUpdateAction(Request $request, $code): Response
+    {
+        $data = $this->getDecodedContent($request->getContent());
+        $data = $this->productModelAttributeFilter->filter($data);
+
+        $productModel = $this->productModelRepository->findOneByIdentifier($code);
+        $isCreation = null === $productModel;
+
+        if ($isCreation) {
+            $this->validateCodeConsistency($code, $data);
+            $productModel = $this->factory->create();
+        } elseif (null !== $productModel->getParent() && array_key_exists('parent', $data) && null === $data['parent']) {
```

In the case someone is trying to update a sub product model with a parent set to null, we should raise an exception here as it won't be raised later because else the product model will be updated with the parent set to null, but the validator won't raise an error as it will think it's a root product model (because it has no parent) and at this point there are no way to know if it was a sub product model or not.

---

You should put it in a dedicated updater of the API.